### DISCUSSION TITLE
✨ Tutorial: How to Buy a Token on the 🦄 Uniswap Universal Router 🦄 with 🐍 Python 🐍 | HackerNoon ✨

### DISCUSSION THREAD

**Discussion by Elnaril** [Elnaril/uniswap-universal-router-decoder#9] [Announcements]
Creating a swap transaction for the latest Uniswap router is not as straightforward as it used to be with the previous 
routers: you cannot just use the router ABI and invoke a swap function. 

This guide will teach you how to use Python to build the transaction input data you need to send to the **Uniswap 
Universal Router** (UR) in order to buy a token with ETH.

✨✨ [Tutorial link](https://hackernoon.com/how-to-buy-a-token-on-the-uniswap-universal-router-with-python) ✨✨

The full tutorial code is [here](https://github.com/Elnaril/uniswap-universal-router-decoder/blob/master/tutorials/buy_tokens/buy_token.py).

Feel free to share any constructive feedback! :)

**Comment by DIMASTA1993** [comment]
Pretty useful!

**Reply by Elnaril** [reply]
Thanks! Really appreciate the feedback! :)

**Comment by Alya0311** [comment]
thy!rly!)

**Comment by Catinpijama** [comment]
Very helpful, thank you!)

**Comment by AceitunaB** [comment]
thanks

**Comment by OliviaBraun** [comment]
Oh, thank you so much for info

**Comment by jack99problems** [comment]
Thank You so much!

**Comment by catheri01** [comment]
I appreciate your help with a guide

**Comment by nadin29091** [comment]
thanks!

**Comment by evgenyshtols** [comment]
Hello everyone good mood

**Comment by sanjalove** [comment]
Hello everyone, I am very glad to be in such a great company.
